src/actions/proxy.ts

```diff
@@ -0,0 +1,161 @@
+import { Constructor, Tinlake } from '../types';
+import { waitAndReturnEvents, executeAndRetry } from '../ethereum';
+const abiCoder = require('web3-eth-abi');
+import BN from 'bn.js';
+import { ethers } from 'ethers';
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+function ProxyActions<ActionsBase extends Constructor<Tinlake>>(Base: ActionsBase) {
+
+  return class extends Base implements IProxyActions {
+
+    getAccessTokenOwner = async (tokenId: string): Promise<BN> => {
+      const res : { 0: BN } = await executeAndRetry(this.contracts['PROXY_REGISTRY'].ownerOf, [tokenId]);
+      return res[0];
+    }
+
+    getNFTOwner = async (tokenId: string): Promise<BN> => {
+      const res : { 0: BN } = await executeAndRetry(this.contracts['COLLATERAL_NFT'].ownerOf, [tokenId]);
+      return res[0];
+    }
+
+    transferNFT = async (from: string, to: string, tokenId: string) => {
+      const txHash = await executeAndRetry(this.contracts['COLLATERAL_NFT'].transferFrom, [from, to, tokenId, this.ethConfig]);
+      console.log(`[NFT Approve] txHash: ${txHash}`);
+      return waitAndReturnEvents(this.eth, txHash, this.contracts['COLLATERAL_NFT'].abi, this.transactionTimeout);
+    }
+
+    buildProxy = async (owner: string) => {
+      const txHash = await executeAndRetry(this.contracts['PROXY_REGISTRY'].build, [owner, this.ethConfig]);
+      console.log(`[Proxy created] txHash: ${txHash}`);
+      const response: any = await waitAndReturnEvents(this.eth, txHash, this.contracts['PROXY_REGISTRY'].abi, this.transactionTimeout);
+      return response.events[0].data[2].toString();
+    }
+
+    getProxy = async (accessTokenId: string) => {
+      const res = await executeAndRetry(this.contracts['PROXY_REGISTRY'].proxies,  [accessTokenId, this.ethConfig]);
+      return res[0];
+    }
+
+    getProxyAccessToken = async (proxyAddr: string) => {
+      const proxy: any = this.eth.contract(this.contractAbis['PROXY']).at(proxyAddr);
+      const res = await executeAndRetry(proxy.accessToken, [this.ethConfig]);
+      return res[0].toNumber();
+    }
+
+    getProxyFromLoan = async (loanId: string) => {
```

Comment:
better naming would be getProxyOwnerFromLoan, since you are not returning the proxy but the proxy owner address  given a loan id